media/src/main/java/com/yas/media/service/MediaServiceImpl.java

```diff
@@ -5,53 +5,53 @@
 import com.yas.media.exception.NotFoundException;
 import com.yas.media.exception.UnsupportedMediaTypeException;
 import com.yas.media.model.Media;
+import com.yas.media.model.dto.MediaDto;
+import com.yas.media.model.dto.MediaDto.MediaDtoBuilder;
+import com.yas.media.repository.FileSystemRepository;
 import com.yas.media.repository.MediaRepository;
+import com.yas.media.utils.StringUtils;
 import com.yas.media.viewmodel.MediaPostVm;
 import com.yas.media.viewmodel.MediaVm;
 import com.yas.media.viewmodel.NoFileMediaVm;
 import java.io.IOException;
 import java.util.Objects;
+import lombok.RequiredArgsConstructor;
 import org.springframework.http.MediaType;
 import org.springframework.stereotype.Service;
 import org.springframework.web.util.UriComponentsBuilder;
 
+@RequiredArgsConstructor
 @Service
 public class MediaServiceImpl implements MediaService {
 
     private final MediaRepository mediaRepository;
+    private final FileSystemRepository fileSystemRepository;
```

Comment:
It should depend on the abstraction rather than the implementation.
In this case, I think we could have an interface like below:
```java
public interface FileStorageService {
    String saveFile(InputStream inputStream, String fileName);
    InputStream getFile(String filePath);
    void deleteFile(String filePath);
}
```
This way, we could easily switch between using `LocalFileStorageService`, `S3FileStorageService`, or another storage provider later.

Reply:
Thanks for your cmt.
But I think we should keep it simple here since it has only one type. 
If we have many types of file storage or requirements said that , we'll consider to spilt this soon.

Reply:
Hi @VanCongBang , even though we currently have only one implementation, I still suggest modifying the code to follow the Dependency Inversion Principle (DIP).
Instead of depending on `FileSystemRepository`, `MediaServiceImpl` should depend on an interface. This way, if a different implementation is introduced, our code will be ready for it with minimal impact.